syntax = "proto2";
package waflz_pb;
import "action.proto";
//: ----------------------------------------------------------------------------
//: bot manager config
//: ----------------------------------------------------------------------------
message bot_manager {
        // -------------------------------------------------
        // properties
        // -------------------------------------------------
        optional string id = 1;
        optional string name = 2;
        optional string customer_id = 3;
        optional string last_modified_date = 4;
        optional string last_modified_by = 5;
        // -------------------------------------------------
        // known bots
        // -------------------------------------------------
        message known_bot {
                optional string bot_token = 1;
                optional enforcement action = 2;
                optional enforcement spoof_action = 3;

        }
        optional bool inspect_known_bots = 50;
        repeated known_bot known_bots = 51;
        //---------------------------------------------------
        // bot rule and action
        //---------------------------------------------------
        optional string bots_prod_id = 60;
        optional enforcement bots_prod_action = 62;
}
